recipe/patches/upstream-PR-367.patch

```diff
@@ -0,0 +1,138 @@
+     lapack_int* info );
+ 
++#define LAPACK_sggsvd LAPACK_GLOBAL(sggsvd,SGGSVD)
++lapack_int LAPACKE_sggsvd( int matrix_layout, char jobu, char jobv, char jobq,
++                           lapack_int m, lapack_int n, lapack_int p,
++                           lapack_int* k, lapack_int* l, float* a,
++                           lapack_int lda, float* b, lapack_int ldb,
++                           float* alpha, float* beta, float* u, lapack_int ldu,
++                           float* v, lapack_int ldv, float* q, lapack_int ldq,
++                           lapack_int* iwork );
++
++#define LAPACK_dggsvd LAPACK_GLOBAL(dggsvd,DGGSVD)
++lapack_int LAPACKE_dggsvd( int matrix_layout, char jobu, char jobv, char jobq,
++                           lapack_int m, lapack_int n, lapack_int p,
++                           lapack_int* k, lapack_int* l, double* a,
++                           lapack_int lda, double* b, lapack_int ldb,
++                           double* alpha, double* beta, double* u,
++                           lapack_int ldu, double* v, lapack_int ldv, double* q,
++                           lapack_int ldq, lapack_int* iwork );
++
++#define LAPACK_cggsvd LAPACK_GLOBAL(cggsvd,CGGSVD)
++lapack_int LAPACKE_cggsvd( int matrix_layout, char jobu, char jobv, char jobq,
++                           lapack_int m, lapack_int n, lapack_int p,
++                           lapack_int* k, lapack_int* l,
++                           lapack_complex_float* a, lapack_int lda,
++                           lapack_complex_float* b, lapack_int ldb,
++                           float* alpha, float* beta, lapack_complex_float* u,
++                           lapack_int ldu, lapack_complex_float* v,
++                           lapack_int ldv, lapack_complex_float* q,
++                           lapack_int ldq, lapack_int* iwork );
++
++#define LAPACK_zggsvd LAPACK_GLOBAL(zggsvd,ZGGSVD)
++lapack_int LAPACKE_zggsvd( int matrix_layout, char jobu, char jobv, char jobq,
++                           lapack_int m, lapack_int n, lapack_int p,
++                           lapack_int* k, lapack_int* l,
++                           lapack_complex_double* a, lapack_int lda,
++                           lapack_complex_double* b, lapack_int ldb,
++                           double* alpha, double* beta,
++                           lapack_complex_double* u, lapack_int ldu,
++                           lapack_complex_double* v, lapack_int ldv,
++                           lapack_complex_double* q, lapack_int ldq,
++                           lapack_int* iwork );
++
+ #define LAPACK_cggsvd3 LAPACK_GLOBAL(cggsvd3,CGGSVD3)
+ void LAPACK_cggsvd3(
+     char const* jobu, char const* jobv, char const* jobq,
+@@ -3679,6 +3742,43 @@ void LAPACK_zggsvd3(
+     lapack_int* iwork,
+     lapack_int* info );
+ 
++#define LAPACK_sggsvp LAPACK_GLOBAL(sggsvp,SGGSVP)
++lapack_int LAPACKE_sggsvp( int matrix_layout, char jobu, char jobv, char jobq,
```

Comment:
Shouldn't this be,
```suggestion
+lapack_int LAPACK_sggsvp( int matrix_layout, char jobu, char jobv, char jobq,
```

Reply:
I didn't touch the upstream patch... This question should better be directed to the folks upstream at Reference-LAPACK/lapack#367.

Reply:
I tend to agree with you (although not knowing much of the codebase), but at least the build seems to be happy with the patch as it is...

Reply:
Since the windows build is failing already (on the tests that are affected by these functions), I'll try. Notably, the linux/osx builds passed somehow.

Reply:
Now that I applied this within the patch, I'm thinking it's maybe clearer to have a separate patch for this...